### DISCUSSION TITLE
Inherit parameters from part templates

### DISCUSSION THREAD

**Discussion by ctft1** [inventree/InvenTree#6123] [Ideas]
Hi,

Is it possible to inherit parameters value from the part template to the linked parts?

It would be useful in some case, example: for a wine seller company, the global "wine producer + wine name" could be the part template (with the wine type/color, region + some other data defined as parameters in the template), and those parameters would be inherited to other "parts" (wines)

It would save a lot of time as only a few parameters should be then set, for example vintage, and bottle size.

This would allow to only keep the template up-to-date for the main parameters, and any change to the template would be propagated to children.

regards,

**Comment by SchrodingersGat** [comment]
Part parameters do not get inherited down the part variant tree in this manner. This would be technically possible but may mean that the parameters cannot be queried from the database in an efficient manner. 

Another option may be to add an option to "update variant parts" when setting the parameter value for a template part. This would effectively copy the parameter values down the tree?

**Comment by ctft1** [comment]
Hi,

Ok, I totally understand this point. Then of course an option to "update variant parts" would do almost the same job (the sync should just be checked from time to time, to ensure the sync is ok even after a propagation failure)

In this case, there could be some additional ideas, for example having an additional setting to determine the behavior of the "copied parameters", it could be : 
- grayed out in "variants" (not editable, always synced with the template value)
- updated from a variant, then propagating to all variants and template
- personalized for some products (different value from the template), in that case it would be easy to only update variants that have the same parameter value for the template and variant
- displaying a list of some "problematic" variants for manual resolution (update or keep existing value)
